base/initdefs.jl

```diff
@@ -266,6 +266,9 @@ function active_project(search_load_path::Bool=true)
         project == "@" && continue
         project = load_path_expand(project)
         project === nothing && continue
+        # while this seems well-inferred, nevertheless without the type annotation below
+        # there are backedges here from abspath(::AbstractString, ::String)
```

Comment:
This seems very strange --- AFAICT we already know that `project` is a `String` here, so I don't know why this would be. We should investigate.

Reply:
Got it --- it's due to recursion via `load_path_expand`. The types get narrowed later, but the edges added before the recursion resolved are still there.